### DISCUSSION TITLE
Turbo Button

### DISCUSSION THREAD

**Discussion by Korkman** [86Box/86Box#4262] [Ideas]
For old machines, CPU frequency could often be changed with a "turbo" button to have extra performance on-demand. In emulation, this would be nice to speed up memory checks and boot time, then slow down to correct speeds for games.

I would like to see a "turbo" setting in the status bar and menus, with a corresponding setting in configuration to set a desired "turbo" speed multiplier in the range of 1.0, 1.5, 2.0, 3.0 ... maybe up to 10.0? 10.0 would be a very unrealistic multiplier, but convenient for example to get past an IBM XT boot process much faster.

**Comment by Mal752** [comment]
I remember the Turbo button fondly in my old systems. They are very useful and nice and I usually turn off the turbo while playing older speed sensitive games on my 286 back then.

But implementing here might be harder, I guess, since the current speed of the systems running in 86Box all are in Turbo ON mode (that is, all the systems are running at the specified CPU speed), and need to introduce lag features to mimic Turbo OFF, and the "slowing down" when set to Turbo OFF, all the required components must be slowed equally and smoothly, which is not easy I guess.

A better option, at least for now, is to run the same system with a lower speed "CPU" or create another lower-spec virtual machine using the 86Box Manager which can maintain multiple "machines" (configurations) to run specific games more comfortably.

**Comment by OBattler** [comment]
There is no need to introduce lag, just to run the CPU with a lower CPU speed if turbo is OFF. This will be done in v2.20.

**Comment by necros2k7** [comment]
Like in Dosbox-X turbo option
